Source/JavaScriptCore/runtime/FunctionRareData.cpp

```diff
@@ -76,8 +76,8 @@ FunctionRareData::FunctionRareData(VM& vm, ExecutableBase* executable)
     , m_executable(vm, this, executable)
     , m_hasReifiedLength(false)
     , m_hasReifiedName(false)
-    , m_hasModifiedLengthForNonHostFunction(false)
-    , m_hasModifiedNameForNonHostFunction(false)
+    , m_hasModifiedLengthForNonNativeHostFunction(false)
+    , m_hasModifiedNameForNonNativeHostFunction(false)
```

Comment:
Per our offline conversation, the reason you added `Native` to the `NonHostFunction` name is because bound functions are also host functions but they are allowed (i.e. not excluded by this non-host filter) as see in `canAssumeNameAndLengthAreOriginal`.  I suggest using `ForBoundOrNonHostFunction` instead to be more accurate.